bin/vcf_neoepitope_predictor.py

```diff
@@ -0,0 +1,354 @@
+                    l = l.strip()
+                    if l != "":
+                        protein_ids.append(l)
+        if args.type == "VEP":
+            variants = read_variant_effect_predictor(args.vcf, gene_filter=protein_ids)
+        elif args.type == "SNPEFF":
+            variants = read_vcf(args.vcf)[0]
+        else:
+            variants = read_annovar_exonic(args.vcf, gene_filter=protein_ids)
+
+        variants = filter(lambda x: x.type != VariationType.UNKNOWN, variants)
+
+        if args.filterSNP:
+            variants = filter(lambda x: x.type != VariationType.SNP, variants)
+
+        if args.filterINDEL:
+            variants = filter(lambda x: x.type not in [VariationType.INS,
+                                                       VariationType.DEL,
+                                                       VariationType.FSDEL,
+                                                       VariationType.FSINS], variants)
+
+        if args.filterFSINDEL:
+            variants = filter(lambda x: x.type not in [VariationType.FSDEL, VariationType.FSINS], variants)
+
+        if not variants:
+            sys.stderr.write("No variants left after filtering. Please refine your filtering criteria.\n")
+            return -1
+
+        epitopes = []
+        for length in range(args.peptide_min_length, args.peptide_max_length):
+            epitopes.extend(filter(lambda x: any(x.get_variants_by_protein(tid) for tid in x.proteins.iterkeys()),
+                                   generate_peptides_from_variants(variants,
+                                                                   length, martDB, EIdentifierTypes.ENSEMBL)))
+
+        for v in variants:
+            for trans_id, coding in v.coding.iteritems():
+                if coding.geneID is not None:
+                    transcript_to_genes[trans_id] = coding.geneID
+                else:
+                    transcript_to_genes[trans_id] = 'None'
+
+    # else: generate protein sequences from given HGNC IDs and than epitopes
+    else:
+        proteins = []
+        with open(args.proteins, "r") as f:
+            for l in f:
+                ensembl_ids = martDB.get_ensembl_ids_from_id(l.strip(), type=EIdentifierTypes.HGNC)[0]
+                protein_seq = martDB.get_product_sequence(ensembl_ids[EAdapterFields.PROTID])
+                if protein_seq is not None:
+                    transcript_to_genes[ensembl_ids[EAdapterFields.TRANSID]] = l.strip()
+                    proteins.append(
+                        Protein(protein_seq, gene_id=l.strip(), transcript_id=ensembl_ids[EAdapterFields.TRANSID]))
+        epitopes = []
+        for length in range(args.peptide_min_length, args.peptide_max_length):
+            epitopes.extend(generate_peptides_from_proteins(proteins, length))
+
+    # read in allele list
+    alleles = read_lines(args.alleles, in_type=Allele)
+
+    result = EpitopePredictorFactory(args.method).predict(epitopes, alleles=alleles)
```

Comment:
we might not need to predict the neoepitopes, as there exist some that are not predicted binders.